### DISCUSSION TITLE
Delay output?

### DISCUSSION THREAD

**Discussion by brucelaughlin** [OpenDrift/opendrift#1266] [Ideas]
I'm not sure how useful this would be to the general particle tracking community, but I'm studying larval connectivity (comparing initial locations with "settlement" locations), and it would be convenient if it were possible to have high output frequency only after a certain amount of time has passed.  This would, for example, allow for high temporal resolution during the "settlement window" of virtual larvae, without needing to generate huge files tracking their entire pelagic journey.  I imagine that allowing for changes in output frequency during a run could be difficult to implement and/or would break CF compliance.  Alternatively, might it be possible to create an option to delay output until a certain condition is met (ie, X days/timesteps have elapsed)?  In this case, perhaps initial conditions could be stored in their own variables in the output file?

Just thinking out loud, as I configure a large experiment under storage constraints!

Cheers,
Bruce

**Comment by knutfrode** [comment]
I believe that a variable time interval would be ok according to the CF-convention.
But it would introduce some relatively fundamental internal changes within OpenDrift, with the danger of inducing problems for most other applications that do not need this feature. But it could be that some small changes in the export module cold achieve this.
https://opendrift.github.io/_modules/opendrift/export/io_netcdf.html#init

A more pragmatic solution in the meantime could perhaps be to first produce the large files as normal, and then directly afterwards delete the unnecessary timesteps to create a smaller file? This should be just a couple of lines with Python/Xarray, or Nco or CDO.
